ui/selectmenu.js

```diff
@@ -210,7 +212,7 @@ return $.widget( "ui.selectmenu", {
 				this._getSelectedItem().data( "ui-selectmenu-item" ) || {}
 			)
 		);
-		if ( !this.options.width ) {
+		if ( this.options.width === null ) {
```

Comment:
The check above is inside `_create()`; the logic is: if we have any size to set, then do so.

This check is inside `refresh()`. If there is no inline style (`false`) or there's an explicit width (anything else beside `null`), then there's no action to take because the state will already be correct. If the value is `null` and we're copying the size of the original element, then we need to resize the button because that's something external that can change the behavior.